src/main/scala/io/iohk/ethereum/vm/OpCode.scala

```diff
@@ -776,7 +779,7 @@ abstract class CallOp(code: Int, delta: Int, alpha: Int) extends OpCode(code, de
         val stack2 = stack1.push(UInt256.Zero)
         val mem2 = mem1.expand(outOffset, outSize)
         val world1 = state.world.combineTouchedAccounts(result.world)
-        val gasAdjustment = if (error == InvalidCall) -startGas else BigInt(0)
+        val gasAdjustment = if (error == InvalidCall) -startGas else if (error == RevertOccurs) -result.gasRemaining else BigInt(0)
```

Comment:
Please look at: https://github.com/input-output-hk/mantis/pull/500/files#diff-cfbbd0cee300114eb59c735b4e84e997R120.

Reply:
You missed second part of my comment : 
`Also in case of REVERT for create op we must handle this return data somehow, the specification states that In case REVERT is used in the context of a CREATE or CREATE2 call, no code is deployed, 0 is put on the stack and the error message is available in the returndata buffer. So in https://github.com/input-output-hk/mantis/pull/493/files#diff-2573bce385123e76cd2b2495911d3b7aL703, there should be something along this lines: val returnData = if (err == RevertOccurs) result.returnData else ByteString.empty and then in https://github.com/input-output-hk/mantis/pull/493/files#diff-2573bce385123e76cd2b2495911d3b7aL706 `

Also i am also little uneasy with handling the memory the same way as in no error case ,it would be safer just extract logic of storing in memory to some other function, and use it only in case of revert and for other errors leave the old way of handling ( just expand memory)